src/pl/senla/hotel/ie/file/ReaderWriterEntity.java

```diff
@@ -0,0 +1,57 @@
+package pl.senla.hotel.ie.file;
+
+import com.opencsv.CSVWriter;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static pl.senla.hotel.constant.InputOutputConstant.DATA_SAVED;
+
+public class ReaderWriterEntity<T> implements ReaderWriterUniversalEntity<T> {
+
+    private final ConverterEntity<T> converter;
+
+    public ReaderWriterEntity(ConverterEntity<T> converter) {
+        this.converter = converter;
+    }
+
+
+    @Override
+    public List<T> load(Class<T> clazz) throws IOException {
```

Comment:
clazz is odd here because you have generic in class definition

Reply:
thank you.